base_layer/core/src/transactions/transaction_protocol/single_receiver.rs

```diff
@@ -111,18 +104,24 @@ impl SingleReceiverTransactionProtocol {
                 .construct_proof(spending_key, sender_info.amount.into())?
         };
 
+        let mut sender_features = sender_info.features.clone();
+        let recovery_byte = OutputFeatures::create_unique_recovery_byte(&commitment, rewind_data);
+        if sender_features.recovery_byte != recovery_byte {
```

Comment:
Nit: this will probably never be true right? If it is, have we saved any work here by doing the compare vs just setting the recovery byte no matter what?

Reply:
Let me simplify this throughout.